doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,229 @@
+any complexity, but does reinforce the single responsibility principle
+among ``Axes.bxp``, ``cbook.boxplot_stats``, and ``Axes.boxplot``.
+
+Additionally, the ``notch`` parameter could be renamed ``shownotches``
+to be consistent with ``Axes.bxp``. This kind of cleanup could be taken
+a step further and the ``whis``, ``bootstrap``, ``autorange`` could
+be rolled into the kwargs passed to the new ``statfxn`` parameter.
+
+Backward compatibility
+======================
+
+Implementation of this MEP would eventually result in the backwards
+incompatible deprecation and then removal of the keyword parameters
+``usermedians``, ``conf_intervals``, and ``sym``. Curosry searches on
+GitHub indicated that ``usermedians``, ``conf_intervals`` are used by
+few users, who all seem to have a very strong knowledge of matplotlib.
+A robust deprecation cycle should provide sufficient time for these
+users to migrate to a new API.
+
+Deprecation of ``sym`` however, may be have a much broader reach into
+the matplotlib userbase.
+
+An accelerated timeline could look like the following:
+
+  1. v2.0.1 add transforms to ``cbook.boxplots_stats``, expose in ``Axes.boxplot``
+  2. v2.1.0 deprecate ``usermedians``, ``conf_intervals``, ``sym`` parameters
+  3. v2.2.0 make deprecations noisier (???)
+  3. v2.3.0 remove ``usermedians``, ``conf_intervals``, ``sym`` parameters
+  4. v2.3.0 deprecate ``notch`` in favor of ``shownotches`` to be consistent with
+     other parameters and ``Axes.bxp``.
+  5. v2.4.0 remove ``notch`` parameter, move all style and artist toggling
+     logic to ``Axes.bxp``. ``Axes.boxplot`` is little more than a broker
+     between ``Axes.bxp`` and ``cbook.boxplots_stats``.
+
+
+Alternatives
+============
+
+This MEP can be divided into a few loosely coupled components:
+
+  1. Allowing pre- and post-computation tranformation function in
+     ``cbook.boxplot_stats``
+  2. Exposing that tranformation in the ``Axes.boxplot`` API
+  3. Removing redundant statistical options in ``Axes.boxplot``
+  4. Shifting all styling parameter processing from ``Axes.boxplot`` to
+     ``Axes.bxp``.
+
+With this approach, #2 depends and #1, and #4 depends on #3.
+
+There are two possible approaches to #2. The first and most direct would
+be to mirror the new ``transform_in`` and ``tranform_out`` parameters of
+``cbook.boxplot_stats`` in ``Axes.boxplot`` and pass them directly.
+
+.. python:
+   fig, ax = plt.subplots()
+   data = mylib.load_data()
+   ax.boxplot(data, ..., transform_in=np.log, transform_out=np.exp)
+
+The second approach would be to add ``statfxn`` and ``statfxn_args``
+parameters to ``Axes.boxplot``. Under this implementation, the default
+value of ``statfxn`` would be ``cbook.boxplot_stats``, but users could
+pass their own function. Then `transform_in`` and ``tranform_out`` would
+then be passed as elements of the ``statfxn_args`` parameter.
+
+Using matplotlib's stats function, this would look similar to this:
+
+.. python:
```

Comment:
To be very annoying, ( :smile: ) could you add an example of how you would do this with the current API?
The difference would be more striking.